examples/post-sync/main.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(v1beta1.AddToScheme(scheme))
+}
+
+// Define a new struct for the unique identifier
+type SyncID struct {
+	Name      string
+	Kind      string
+	Namespace string
+}
+
+type SyncStatusController struct {
+	client        client.Client
+	log           logr.Logger
+	statusTracker *StatusTracker
+	syncKind      string // Explicit field to indicate what kind of resource we're watching
+}
+
+func (c *SyncStatusController) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
```

Comment:
suggest pulling the controller code out to its own file and adding some integration tests that use the controller-manager. There's some examples in config sync controller that set up the controller manager.

Reply:
Controller pulled out to a separate file, still in the same package to get around the import issue. Also added unit test leveraging the statustracker. Working on other refinements now.